pandas/core/arrays/sparse/array.py

```diff
@@ -701,7 +701,11 @@ def isna(self):
         # If null fill value, we want SparseDtype[bool, true]
         # to preserve the same memory usage.
         dtype = SparseDtype(bool, self._null_fill_value)
-        return type(self)._simple_new(isna(self.sp_values), self.sp_index, dtype)
+        if self._null_fill_value:
+            return type(self)._simple_new(isna(self.sp_values), self.sp_index, dtype)
```

Comment:
do we still need to be concerned about nulls in the sp_values?

Reply:
I agreed that is looks redundant, but now I can reproduce the case when without this check the result will be wrong.
It related with [the issue](https://github.com/pandas-dev/pandas/issues/44956).
~~~python
s = pd.arrays.SparseArray([np.nan, 1,2,3,0,0,0, np.nan], fill_value=0)
#[nan, 1.0, 2.0, 3.0, 0, 0, 0, nan]
#Fill: 0
#IntIndex
#Indices: array([0, 1, 2, 3, 7])
s.fill_value = np.nan
s
#[nan, 1.0, 2.0, 3.0, nan, nan, nan, nan]
#Fill: nan
#IntIndex
#Indices: array([0, 1, 2, 3, 7])

pd.isna(s)
#[True, False, False, False, True, True, True, True]
#Fill: True
#IntIndex
#Indices: array([0, 1, 2, 3, 7])
~~~

By the way, for this case indices of last array are worng. To be fixed and tested...

Reply:
> s.fill_value = np.nan

I didn't realize this was settable.  Should definitely deprecate that.

I guess without `__setitem__` its tough to get cases where you have both a nan fill_value and nan in the sp_values?

Reply:
Setting of fill_value doesn't use __setitem__. E.g. for mentioned case we can't use __setitem__:

~~~python
 s[0] = 0
---------------------------------------------------------------------------
TypeError: SparseArray does not support item assignment via setitem
~~~

I'm not sure is this correct to deprecate changing fill_value? 
Perhaps we have to add some value checking depending from type and nan...